### PR TITLE
blaspp: set BLA_VENDOR for cray

### PR DESCRIPTION
**Pull request by lukebroskop**

### CODE REVIEW

**Review by teonnik — DISMISSED** [review]

**Review by Sely85 — DISMISSED** [review]

**Comment by mgates3** [comment]
I don't understand this PR. Spack is requesting to use a specific BLAS library (Cray libsci), but then this is circumventing that to do a general search using BLA_VENDOR=all (where "all" is not even a listed option at https://cmake.org/cmake/help/latest/module/FindBLAS.html#blas-lapack-vendors). The search may find Cray libsci, but it may find something else like OpenBLAS — just depends on what is available. Can @lukebroskop please describe what this was suppose to be fixing & how?

We can investigate more after the holidays and submit a correction.

**Review by sethrj — CHANGES_REQUESTED** [review]
Same as https://github.com/spack/spack/pull/28125: instead of adding empty flags, make `args` a local variable and `args.append(self.define('BLA_VENDOR', 'all'))`.

Also are you sure it should be `all` instead of `cray`? because that's the one you require in `lapackpp`.

**Comment by lukebroskop** [comment]
Thanks @mgates3 , I used `BLA_VENDOR=all` to work around a cmake configure error on spock. I will change `BLA_VENDOR=all` to `BLA_VENDOR=CRAY` which should have the same effect. Do you have another suggestion?

**Comment by lukebroskop** [comment]
Thanks @sethrj

File: var/spack/repos/builtin/packages/blaspp/package.py [hunk 1]
```diff
@@ -53,12 +53,18 @@ def cmake_args(self):
...
+            args+='-DBLA_VENDOR=CRAY '
+
         return [
             '-Dbuild_tests=%s'       % self.run_tests,
             '-Duse_openmp=%s'        % ('+openmp' in spec),
             '-DBUILD_SHARED_LIBS=%s' % ('+shared' in spec),
             backend_config,
-            '-DBLAS_LIBRARIES=%s'    % spec['blas'].libs.joined(';')
+            '-DBLAS_LIBRARIES=%s'    % spec['blas'].libs.joined(';'),
+            args
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/blaspp/package.py]: This is not right, and you'll also note the style tests are failing due to lines 59 and probably for the superfluous parens in 58. Do:
```
        args = [
            '-Dbuild_tests=%s'       % self.run_tests,
            '-Duse_openmp=%s'        % ('+openmp' in spec),
            '-DBUILD_SHARED_LIBS=%s' % ('+shared' in spec),
            backend_config,
            '-DBLAS_LIBRARIES=%s'    % spec['blas'].libs.joined(';'),
        ]

        if spec['blas'].name == 'cray-libsci':
            args.append(self.define('BLA_VENDOR', 'CRAY'))

        return args
```

**Review by sethrj — CHANGES_REQUESTED** [review]

**Review by sethrj — APPROVED** [review]

**Review by teonnik — APPROVED** [review]
I tested this on Piz Daint with `cray-libsci` explicitly set as a dependency and it builds fine.

**Pull request merged**